libs/langchain/langchain/tools/graphql/tool.py

```diff
@@ -21,21 +21,51 @@ class BaseGraphQLTool(BaseTool):
     If an error is returned with 'Bad request' in it, rewrite the query and try again.
     If an error is returned with 'Unauthorized' in it, do not try again, but tell the user to change their authentication.
 
-    Example Input: query {{ allUsers {{ id, name, email }} }}\
+    {example}
+
+    {schema}\
     """  # noqa: E501
 
+    description_example = "Example Input: query {{ allUsers {{ id, name, email }} }}"
+    description_schema = "Use the following schema for your queries: \n    {schema}\n"
+
+    def __init__(self, **kwargs: Any):
+        super().__init__(**kwargs)
+        self.graphql_wrapper = kwargs["graphql_wrapper"]
+        self.add_schema_to_description()
+
     class Config:
         """Configuration for this pydantic object."""
 
         arbitrary_types_allowed = True
 
+    def add_schema_to_description(self) -> None:
+        """Automatically called if we decided to load the schema.
+        Schema is included it in the prompting to the model"""
+        if self.graphql_wrapper.disable_schema_prompt:
+            self.description = self.description.format(
+                example=self.description_example, schema=""
+            )
+        else:
+            raw_schema = self.graphql_wrapper.gql_schema
+            raw_schema = raw_schema.replace("{", "{{")
```

Comment:
Two questions here: 

why does the code replace `{` with `{[`?

the description seems to contain leading whitespace. Could we remove it to conserve token counts?

Reply:
Graphql uses "{" as part of the schema def, but this is also the format for string formats in python. So i am escaping all the instance of this so when the formatting is applied its usable.

Reply:
If I understand correctly, the code is interpolating the schema into the description (makes sense). Are we formatting the contents of the schema itself anywhere?

Reply:
Yes, when the wrapper pulls the schema through the introspection schema it runs a "printer" that exports it to a nice format: 

```
 values["gql_schema"] = print_filtered_schema(
            build_client_schema(intros_result),
            ```